lib/ansible/galaxy/dependency_resolution/providers.py

```diff
@@ -186,11 +189,11 @@ def find_matches(self, requirements):
                 if all(self.is_satisfied_by(requirement, candidate) for requirement in requirements)
                 # FIXME
                 # if all(self.is_satisfied_by(requirement, candidate) and (
-                #     requirement.src is None or
+                #     requirement.src is None or  # if this is true for some candidates but not all it will break key param - Nonetype can't be compared to str
```

Comment:
I'm curious if that `for requirement in requirements` on line 183 should be removed.

Reply:
Good catch, thanks

Reply:
I think I've dropped it already.